### ISSUE TITLE
Compatibility for ROMs which extend the Resources class themselves

### ISSUE THREAD

**Issue by airani** [rovo89/XposedInstaller#178]
Hi, 
My phone is Huawei P6 with android 4.4.2 and when i upgrade my Xposed to new version (2.6) that is installed but not activated and not worked!
Please fix this bug!

**Comment by pylerSM** [comment]
@rovo89 Some user reported that 2.4.1 xperia test works on Huawei devices.
http://forum.xda-developers.com/xposed/modules-framework-t2991683#post57920507

**Comment by rovo89** [comment]
That's not surprising me at all. Up to 2.5.1, it should work with the regular versions - well, at least Xposed loads, ROM-specific theming functions get lost. On other ROMs, it was crashing completely.

The problem is: Both the ROM and Xposed replace Resources with a specific subclass. But only one can win. Before 2.6, Xposed won, with the effect in parts of the ROM where the ROM-specific subclass is expected, the system crashes. Also some ROMs just add new fields to Resources, which I had to copy and manually copy over to the XResources instance. The new approach is generally safer and works fine for ROMs just adding fields, but Xposed detects conflicts and produces error messages as displayed here.

As a general workaround, resource hooks can be disabled in the settings. For Huawei, I believe someone found out that changing ro.config.hwtheme will avoid that the ROM tries to use its own subclass, so there's no conflict and Xposed works fine. No idea how well that works and if there are any side-effects (except that the ROM's theming engine won't work).

Basically, the solution would be that XResources extends the ROM's class instead of the Resources class directly. I have created such a hack for some MIUI version, but there are quite a few ROMs which would need this and especially Huawei sometimes needs this, but sometimes it would be counterproductive (depending on that parameter mentioned above). I thought about dynamically patching the XposedBridge dex on the ROM, but the specifications are too strict, it's much effort to change it without recompiling. I have another idea, involving an additional jar file with the specific intermediate class per ROM, but it's rather complicated as well, so I don't know when I will be able to try this.

**Comment by mieik** [comment]
Here with an EMUI 3.0 ROM (Honor 6). Installing a precedent version of Xposed Framework and editing the build.prop file leads to have the framework working without disabling the hooks. Unfortunately, modules don't work. I tried GravityBox and it makes no changes; I tried minminguard and the ads are still there. Has anyone came up with a solution to make these modules work? :(

**Comment by iqbalmuhamad** [comment]
installing unsigned apps not working huawei y33? help me

**Comment by anaschillin** [comment]
@rovo89 Did you ever figure out how a global patch for rom that use their own theme engine?

**Comment by ArionMiles** [comment]
@rovo89 Can you at least write a guide on how we can have the Framework work without the need to disable Resource Hooks on some devices?

**Comment by rovo89** [comment]
Since one of the recent versions (85 or 86), this has been fixed.

**Comment by ArionMiles** [comment]
Forgive my noobish question but, I'm using v78. Where can I download the newer version in which this has been fixed? Also, are all the xposed frameworks universally compatible with all androids, i.e., will I brick my device if I upgrade? 

My xposed installer doesn't notify me of an update. I'm on Lollipop.

Thanks!

**Comment by cubakcabak** [comment]
i guess its fixed for newer android version, sadly its not compatible for kitkat anymore.
